packages/bundler/src/BundlerCollectorTracer.ts

```diff
@@ -214,12 +233,19 @@ export function bundlerCollectorTracer (): BundlerCollectorTracer {
         return
       }
 
+      const lastOpInfo = this.lastThreeOpcodes[this.lastThreeOpcodes.length - 2]
       // store all addresses touched by EXTCODE* opcodes
-      if (opcode.match(/^(EXT.*)$/) != null) {
-        const addr = toAddress(log.stack.peek(0).toString(16))
+      if (lastOpInfo?.opcode?.match(/^(EXT.*)$/) != null) {
+        const addr = toAddress(lastOpInfo.stackTop3[0].toString(16))
         const addrHex = toHex(addr)
+        const last3opcodesString = this.lastThreeOpcodes.map(x => x.opcode).join(' ')
         // only store the last EXTCODE* opcode per address - could even be a boolean for our current use-case
-        this.currentLevel.extCodeAccessInfo[addrHex] = opcode
+        if (last3opcodesString.match(/^(\w+) EXTCODESIZE ISZERO$/) == null) {
+          this.currentLevel.extCodeAccessInfo[addrHex] = opcode
```

Comment:
we need to ignore this opcode (EXTCODESIZE) if the contract has code.
- latest opcode is ISZERO, and the STACK holds the size, so at this point, simply ignore the opcode if topOfStack!=0
- ext* counting should be done after this (and always on the one-before-last opcode in lastTheeOpCodes)
lastOpcodes should be reversed, so lastOpcodes[0] is the last, and lastOpcodes[1] is the one before.. this way, we're not fixing its size. currently we need 2, but in the future we might make it longer, and better not have the length as part of its name..